tests/Testing/Concerns/InteractsWithDatabaseTest.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Illuminate\Tests\Testing\Concerns;
+
+use Illuminate\Database\ConnectionInterface;
+use Illuminate\Database\Query\Expression;
+use Illuminate\Database\Query\Grammars\MySqlGrammar;
+use Illuminate\Database\Query\Grammars\PostgresGrammar;
+use Illuminate\Database\Query\Grammars\SQLiteGrammar;
+use Illuminate\Database\Query\Grammars\SqlServerGrammar;
+use Illuminate\Foundation\Testing\Concerns\InteractsWithDatabase;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Facade;
+use Mockery as m;
+use PHPUnit\Framework\TestCase;
+
+class InteractsWithDatabaseTest extends TestCase
+{
+    protected function setUp(): void
+    {
+        Facade::clearResolvedInstances();
+        Facade::setFacadeApplication(null);
+    }
+
+    protected function tearDown(): void
+    {
+        m::close();
+    }
+
+    public function testCastToJsonSqlite()
+    {
+        $grammar = new SQLiteGrammar();
+
+        $this->assertEquals(<<<'TEXT'
```

Comment:
is there a reason you're using Nowdocs everywhere in the tests? Could you just use a simple string here to make it more readable?

```php
$this->assertEquals('["foo", "bar"]', $this->castAsJson(['foo', 'bar'], $grammar));
```

Reply:
Shoot your right -- thanks @browner12, I'll update this 👍

Reply:
Scratch that -- forgot these are intended, because the actual output of the string contains the quoted string. Using `$this->assertEquals('["foo", "bar"]', ...)` asserts that the output is `["foo", "bar"]`, not the fully quoted `'["foo", "bar"]'`.

Reply:
gotcha, thanks for clarifying.